app/app/calculations/punching.py

```diff
@@ -0,0 +1,212 @@
+    # Element = 'plyta'    # do wyboru: plyta, stopa fundamentowa TODO
+    cx = float(b) * cm     # cx = 40 * cm
+    cy = float(h) * cm     # cy = 40 * cm
+    #cxy = 40 * cm	       # TODO(LZ): dla przekroju kolowego
+    dsw = dsw.value * mm   # srednica zbrojenia na ścinanie
+    dx = float(dx) * cm
+    dy = float(dy) * cm
+    lx = float(lx) * cm    # odleglosc pola obciazenia do krawedzi plyty w kier. x
+    ly = float(ly) * cm    # jw. w kier. y
+    ad = float(ad) * cm
+    asx = float(asx) * cm2
+    asy = float(asy) * cm2    
+    lambda_u = lambda_u    # wsp. uwzgledniajacy otwor w poblizu obwodu kontrolnego (%)  # TODO: to przypisanie jest tu bez sensu, ale zrobiłam tak, żeby był ślad po lambda_u
+    alfa = math.pi / 2     # = 90deg, kat miedzy zbrojeniem na ścinanie i plaszczyzna plyty
+
+    # -----------------------------------------
+    # WARUNKI PRACY
+    # Faktory
+    design_situation_id = design_situation.id
+    if design_situation_id == 0 or design_situation_id == 1: # 'trwała' lub 'przejściowa'
+        gamma_c = 1.4
+        gamma_s = 1.15
+        alfa_cc = 1.0
+    elif design_situation_id == 2:                       # 'wyjątkowa'
+        gamma_c = 1.2
+        gamma_s = 1.0
+        alfa_cc = 1.0
+    else:
+        ERRORS.append('Błąd w określeniu sytuacji obliczeniowej.')
+        gamma_c = 0.0
+        gamma_s = 0.0
+        alfa_cc = 0.0
+
+    # Obciazenia
+    beta = float(beta)		# gdzie jest beta?
+    ved = float(ved)
+    ved = ved * kN  # 600 * kN	# dodac komentarz, ze nie jest to reakcja, tylko sila tnaca z plyty na slup
+                    # dla fundamentow mozna zredukowac sile do VEdred - por. 6.4.4(2)
+
+    # ############## OBLICZENIA ##############
+    # -----------------------------------------
+    # MATERIAL
+    fcd = alfa_cc * fck / gamma_c
+    fyd = fyk / gamma_s
+    fywd = alfa_sw * fyk
+
+    # -----------------------------------------
+    # GEOMETRY
+    d = (dx + dy) / 2 - ad  # spr. czy sie nie koliduje z 'D' (srednica kola)
+
+    # TODO(LZ): Dodac warunek na section i powiazac support
+
+    # TODO: ściana - naroże (3) i ściana - koniec (4)
+    support_id = support.id
+    if support_id == 0:     # słup wewnętrzny
+        u00 = 2 * (cx + cy)
+    elif support_id == 1:   # słup krawędziowy X
+        u00 = min(2 * cx + cy, cy + 3 * d)
+    elif support_id == 2:   # słup krawędziowy Y
+        u00 = min(cx + 2 * cy, cx + 3 * d)
+    elif support_id == 3:   # słup narożny
+        u00 = min(cx + cy, 3 * d)
+    elif support_id == 4:   # ściana - naroże
+        u00 = 100           # TODO(LZ): poprawic!
+        ERRORS.append("Podpora 'ściana-naroże' niedostępna.")
+    elif support_id == 5:   # ściana - koniec
+        u00 = 200           # TODO(LZ): poprawic!
+        ERRORS.append("Podpora 'ściana-koniec' niedostępna.")
+    else:
+        # TODO(LZ): dodac sciany (naroznik, koniec)
+        ERRORS.append('Błąd w określeniu położenia podpory (u0).')
+        u00 = 0.0
```

Comment:
Gdzieś uciekły te moje wyciągnięte funkcje. Nie za bardzo mi się to podoba, bo zaciemnia kod.